BSTSuccessor.java

```diff
@@ -0,0 +1,61 @@
+public class BSTSuccessor {
+    public static void main(String[] args) {
+        test(new int[] {50, 25, 75, 12, 37, 62, 87, 6, 18, 31, 43, 56, 68, 81, 93}, new int[]{0, 31, 26, 50, 100, 76, 84});
+
+    }
+
+
+    static void test(int[] input, int[] test) {
+        Node bstRoot = createBST(input);
+        for (int i = 0; i < test.length; i++) {
+            System.out.println("Successor of " + test[i] + " is " + findPredecessor(bstRoot, test[i]));
+        }
+    }
+
+
+    static int findPredecessor(Node bstRoot, int key) {
+        if(bstRoot == null || bstRoot.data == key) return -1;
```

Comment:
When `bstRoot.data == key` you cannot return -1 since there can be a valid successor in the right sub tree for that node.